woocommerce/woocommerce-admin
# Onboarding: Add products task

Fixes #2479 

* Adds the products task.
* Adds task completion checks to `wcSettings`.
* Marks items complete in the task list if complete (currently only on initial page load- follow-up needed to fetch async).
* Fixes the task list item `className` prop.

### Screenshots
<img width="925" alt="Screen Shot 2019-07-18 at 6 50 38 PM" src="https://user-images.githubusercontent.com/10561050/61452764-1f2d7100-a98f-11e9-8f72-5f8161ee7a2a.png">

### Detailed test instructions:

1. Update `const requiredTasksComplete = false;` in dashboard/index.js.
2. Visit the dashboard.
3. Click on the "Products" task item.
4. Make sure each of the sub-task items goes to the correct page.
5. Make sure that after completing the task (i.e., adding a product) you are redirected back to the task list dashboard.
6. Make sure the task list "Products" item is marked completed if you have any products.

**Note:** The redirect back to the task list is only enabled if you had 0 products to begin with.  To test, either use a site with 0 products or comment out the following lines in `set_active_task()`:
```php
			if ( self::check_task_completion( $task ) ) {
				return;
			}
```
